WalletWasabi/BitcoinP2p/P2pBehavior.cs

```diff
@@ -76,8 +76,10 @@ private async Task ProcessInventoryAsync(Node node, InvPayload invPayload)
 		{
 			if (ProcessInventoryVector(inv, node.RemoteSocketEndpoint))
 			{
-				var newInv = new InventoryVector(InventoryType.MSG_WTX, inv.Hash);
-				getDataPayload.Inventory.Add(newInv);
+				if (inv.Type is InventoryType.MSG_TX or InventoryType.MSG_WTX or InventoryType.MSG_BLOCK)
```

Comment:
We are not interested in any other kind of data, only transactions and blocks.

Reply:
I think this is not the right place to check. You see `ProcessInventoryVector` method above is an abstract one, implemented by the implementors of the `P2pBehavior`, like `TrustedP2pBehavior` and `UntrustedP2pBehavior`.